### PR TITLE
Add (Region)Autoscaler

### PR DESCRIPTION
**Pull request by danawillow**
Adds Autoscaler and RegionAutoscaler to Magic Modules.

For now, I've excluded them from the other providers because there's still a little bit of work to be done there around the custom `update_url` (and we'll want to decide if we want to release RegionAutoscaler before RegionInstanceGroupManager exists, and potentially other stuff), but I'd also be ok with going in and making those changes if you think it's important to do now

Other things this does besides just raw generation:
* Adds an `update_url` field for resources that support PUTting the whole resource but have a custom URL involved in that.
* Adds a new custom Terraform expander for values that have to be URL-shaped
* Fixes a bug in the Terraform templates where nested fields weren't getting renamed
* Copies the Terraform autoscaler tests here because there were some changes I needed to make around the id format.

-----------------------------------------------------------------
# [all]
## [terraform]
Add generated google_compute_autoscaler and google_compute_region_autoscaler resources. Because this adds the ability to import by more than just the name, we can get rid of all the code in resource_compute_autoscaler.go that tries to find the autoscaler in the provider-specified region.

### CODE REVIEW

File: templates/terraform/resource.erb [hunk 1]
```diff
@@ -322,7 +322,7 @@ func resource<%= resource_name -%>Update(d *schema.ResourceData, meta interface{
 	<% elsif object.custom_code.encoder -%>
 	obj, err = resource<%= resource_name -%>Encoder(d, meta, obj)
 	<% end -%>
-	url, err := replaceVars(d, config, "<%= self_link_url(object) -%>")
+	url, err := replaceVars(d, config, "<%= update_url(object, object.update_url) -%>")
```
Comment (ndmckinley) [inline templates/terraform/resource.erb]: I think this logic belongs in core, and should be in PCA also.  It does have to get done next quarter if not today.

File: api/resource.rb [hunk 2]
```diff
@@ -26,6 +26,7 @@ module Properties
       attr_reader :description
       attr_reader :kind
       attr_reader :base_url
+      attr_reader :update_url
```
Comment (ndmckinley) [inline api/resource.rb]: Can you add a comment here explaining what this is for?  And add it to the spreadsheet of resource attributes?

Comment (danawillow) [reply api/resource.rb]: Done and done

File: templates/terraform/resource.erb (hunk 1 above)
Comment (danawillow) [reply templates/terraform/resource.erb]: Fair. In the interest of keeping PRs smallish, I'm not going to do it now (doing so would merit adding Autoscaler to the other providers to show that it works, which means adding examples and testing to make sure they actually work. this is non-trivial enough that I'd like to take my time on it)

**Pull request merged**